options/options.go

```diff
@@ -245,6 +245,10 @@ type TerragruntOptions struct {
 	// in this list.
 	ModulesThatInclude []string
 
+	// When used with `run-all`, restrict the units in the stack to only those that read at least one of the files
+	// in this list.
+	UnitsReading []string
```

Comment:
I think clone() function should be updated to copy conents

Reply:
Can be used a map to store files, and avoid iterations to find

Reply:
Can you elaborate on how you'd want that implemented?